### PR TITLE
feat(volar): add `@vue-macros/volar` plugin

### PR DESCRIPTION
**Pull request by zhiyuanzmj**

### CODE REVIEW

**Comment by zhiyuanzmj** [comment]
It's done.

File: packages/volar/tsup.config.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { resolve } from 'node:path'
 import { defineConfig } from 'tsup'
 
 export default defineConfig({
-  entry: ['./src/*.ts', '!./src/*.d.ts', '!./src/common.ts'],
```
Comment (sxzz) [inline packages/volar/tsup.config.ts:6]: Why remove the exclusion?

Comment (zhiyuanzmj) [reply packages/volar/tsup.config.ts:6]: I only remove the `'!./src/common.ts'` to export common function for my own Volar plugin.

Comment (zhiyuanzmj) [reply packages/volar/tsup.config.ts:6]: After Volar 2.0 a lot of TS native function is disabled.
https://github.com/volarjs/volar.js/issues/165

So I have to write the alternative solution for my each Volar plugin, 
https://github.com/zhiyuanzmj/volar-plugin-ignore-attributes/blob/main/src/index.ts#L96
https://github.com/zhiyuanzmj/unplugin-vue-reactivity-function/blob/main/src/volar.ts#L147

**Review by sxzz — APPROVED** [review]

**Pull request merged**